Code/Yospace/Source/YospaceID3MetadataHandler.swift

```diff
@@ -0,0 +1,95 @@
+//
+//  YospaceID3MetadataHandler.swift
+//
+//  Created by Raffi on 30/05/2024.
+//  Copyright © 2024 THEOplayer. All rights reserved.
+//
+
+import THEOplayerSDK
+import YOAdManagement
+
+class YospaceID3MetadataHandler {
+    private let player: THEOplayerSDK.THEOplayer
+    private let session: YOSession
+    private var addTrackEventListener: THEOplayerSDK.EventListener?
+    private var addCueEventListener: THEOplayerSDK.EventListener?
+    private var metadata: YospaceTimedMetadata = .init()
+
+    init(player: THEOplayerSDK.THEOplayer, session: YOSession) {
+        self.player = player
+        self.session = session
+        self.addTrackEventListener = self.player.textTracks.addEventListener(type: THEOplayerSDK.TextTrackListEventTypes.ADD_TRACK) { [weak self] event in
+            self?.addTrackEventHandler(event: event)
+        }
+    }
+
+    private func addTrackEventHandler(event: THEOplayerSDK.AddTrackEvent) {
+        if let textTrack: THEOplayerSDK.TextTrack = event.track as? THEOplayerSDK.TextTrack,
+           textTrack.kind == THEOplayerSDK.TextTrackKind.metadata._rawValue,
+           textTrack.type == "id3" {
+            self.addCueEventListener = textTrack.addEventListener(type: THEOplayerSDK.TextTrackEventTypes.ADD_CUE) { [weak self] event in
+                self?.addCueEventHandler(event: event)
+            }
+        }
+    }
+
+    private func addCueEventHandler(event: THEOplayerSDK.AddCueEvent) {
+        if let key: String = event.cue.contentDictionary?["id"],
+           let value: String = event.cue.contentDictionary?["text"] {
+            let _value: String = String(value[value.index(after: value.startIndex)..<value.endIndex])
+            if key == "YMID" {
+                if !self.metadata.ymid.isEmpty {
+                    self.metadata = YospaceTimedMetadata()
+                }
```

Comment:
I'm not sure if it's safe to assume that `YMID` will always appear first in the ID3 tag. In the other Yospace connectors, we can handle them in any order.

Reply:
The YMID is the `mediaId`, so it makes sense that it comes first. But, I already have a reset mechanism right after the dispatch, so this code block can be safely removed.